### PR TITLE
Clean up code base, add pre-commit, add dependencies

### PR DESCRIPTION
**Pull request by Zeitsperre**
Addresses issues raised in #2, but does not close it.

Ping @agstephens @sol1105 

### Changes

* Adds a pre-commit configuration for automated code linting
* NetCDF grid downloader has been cleaned up for readability (Note: Cleaning performed with the aid of ChatGPT)
* Base Python has been changed to Python3.9 (follows from [NEP-0029](https://numpy.org/neps/nep-0029-deprecation_policy.html))

### Work needed

1. There are a few instances where CLISOPS is called. If this is to become a dependency of CLISOPS, we need to be implementing the functions required here to prevent a circular dependency loop.
2. The reference grid files are very large! If we only require the grids themselves, it's preferable that the variable values as well as the time dimensions be removed from all files. At the current size, this package cannot be hosted with its data on PyPI.

### Proposal

Would this library better serve as a module of CLISOPS instead? Are there instances where the tools here would be used independently of CLISOPS? Given the tight focus and coupling with CLISOPS, this seems more like a specialized helper library. If so, we can very easily push this directly into CLISOPS.

### CODE REVIEW

File: roocs_grids/scripts/create_grid_file.py [hunk 1]
```diff
@@ -2,6 +2,9 @@
 from pathlib import Path
 
 import xarray as xr
+
+# FIXME: This is a circular dependency, as clisops will be listing roocs-grids as a dependency
```
Comment (sol1105) [inline roocs_grids/scripts/create_grid_file.py:6]: The `scripts` directory is basically a directory with auxiliary scripts to prepare the grid files. None of its contents are imported by the roocs_grids module. I don't know if there is a proper approach for something like this. Maybe one could rename it to `aux` and move it one directory up, next to `tests`, `roocs_grids` so it won't get accidentally imported?

File: roocs_grids/scripts/create_grid_file.py [hunk 2]
```diff
@@ -30,6 +33,7 @@
 # Create clore.Grid object
 g = clore.Grid(ds)
 # Drop variables and global attributes
+# FIXME: This is broken in newer versions of xarray
```
Comment (sol1105) [inline roocs_grids/scripts/create_grid_file.py:36]: I will need to have a look in the clisops regridding PR when time allows, to solve this.

File: roocs_grids/scripts/retrieve_grids.sh [hunk 3]
```diff
@@ -113,58 +102,54 @@ for nc in  ${CMIP6Grids[@]} ${ATLASGrids[@]}; do
...
 }
-[[ ! -e $targetpath/grid_T63_lsm_binary.nc ]] && {
-    mv $targetpath/sftlf_fx_MPI-ESM1-2-LR_1pctCO2_r1i1p1f1_gn.nc $targetpath/grid_T63_lsm_binary.nc
-    ncks -A -h -v areacella $targetpath/areacella_fx_MPI-ESM1-2-LR_1pctCO2_r1i1p1f1_gn.nc $targetpath/grid_T63_lsm_binary.nc
-    nccopy -d 9 -s $targetpath/grid_T63_lsm_binary.nc $targetpath/grid_T63_lsm_binary_def.nc
-    mv $targetpath/grid_T63_lsm_binary_def.nc $targetpath/grid_T63_lsm_binary.nc
+[[ ! -e "$targetpath"/grid_T63_lsm_binary.nc ]] && {
+    mv "$targetpath"/sftlf_fx_MPI-ESM1-2-LR_1pctCO2_r1i1p1f1_gn.nc "$targetpath"/grid_T63_lsm_binary.nc
+    ncks -A -h -v areacella $targetpath/areacella_fx_MPI-ESM1-2-LR_1pctCO2_r1i1p1f1_gn.nc "$targetpath"/grid_T63_lsm_binary.nc
+    nccopy -d 9 -s "$targetpath"/grid_T63_lsm_binary.nc "$targetpath"/grid_T63_lsm_binary_def.nc
```
Comment (sol1105) [inline roocs_grids/scripts/retrieve_grids.sh:130]: I will need to extend the compression to all downloaded grid files and remove unnecessary attributes, variables and dimensions. By just applying `nccopy -d 9 -s` to all the grid files, the volume could be shrinked from 45M to 2.4M, which would be fine? The volume was one of the reasons, `roocs_grids` was a separate repository and not part of the clisops regridding PR.

I will follow up your PR with another one dealing with the compression and further cleaning of the netcdf files.

**Review by sol1105 — APPROVED** [review]
This looks great, thank you.
I will deal with the `#FIXME` points once this PR is merged.

**Review by cehbrecht — APPROVED** [review]

**Comment by Zeitsperre** [comment]
@sol1105 

Your comments sound good. I'll merge this now and leave you to work on the comments raised here.

All the best!

**Pull request merged**